src/main/java/uk/gov/gchq/palisade/resource/AbstractLeafResource.java

```diff
@@ -87,22 +114,12 @@ public Boolean isAttributeSet(final String attributeKey) {
         return this.attributes.containsKey(attributeKey);
     }
 
-    @Override
-    public void setType(final String type) {
-        type(type);
-    }
-
-    @Override
-    public void setSerialisedFormat(final String serialisedFormat) {
-        serialisedFormat(serialisedFormat);
-    }
-
-    public void setAttributes(final Map<String, Object> attributes) {
-        attributes(attributes);
-    }
 
     public void setAttribute(final String attributeKey, final Object attributeValue) {
-        attribute(attributeKey, attributeValue);
+        requireNonNull(attributeKey, "The attributeKey cannot be set to null.");
+        requireNonNull(attributeKey, "The attributeValue cannot be set to null.");
+        this.attributes.put(attributeKey, attributeValue);
+
     }
 
     public AbstractLeafResource parent(final ParentResource parent) {
```

Comment:
This needs the @generated annotation on the mutator and the method needs updating to remove the null check and to use the setter method

Reply:
Done and moved to sit with the other mutators.